scalyr_agent/builtin_monitors/journald_monitor.py

```diff
@@ -435,3 +559,169 @@ def gather_sample(self):
             self._checkpoint.update_checkpoint(
                 self._checkpoint_name, str(self._last_cursor)
             )
+
+    def stop(self, wait_on_join=True, join_timeout=5):
+        ScalyrMonitor.stop(self, wait_on_join=wait_on_join, join_timeout=join_timeout)
+        self.log_manager.close()
+
+
+def _create_log_config(
+    base_config, matched_config, global_config, file_template, match_hash=None
+):
+    log_config = base_config.copy()
+    attrs = log_config.get("attributes", JsonObject({}))
+    log_config.update(matched_config)
+    # Set the parser the log_config['parser'] level
+    # otherwise it will be overwritten by a default value due to the way
+    # log_config verification works
+    log_config["parser"] = get_parser_from_config(matched_config, attrs, "journald")
+
+    attrs.update(matched_config.get("attributes", JsonObject({})))
+    log_config["attributes"] = attrs
+    if match_hash:
+        full_path = os.path.join(
+            global_config.agent_log_path,
+            file_template.safe_substitute({"ID": match_hash}),
+        )
+        log_config["path"] = full_path
+
+    return log_config
+
+
+class LogManager:
+    def __init__(
```

Comment:
After our discussion, I believe the only parameters should be `global_config`, `formatter`, `max_log_size`, `max_log_rotations`.  (`formatter` is used when creating a logger`.